### ISSUE TITLE
Too many accounts corrupts database?

### ISSUE THREAD

**Issue by shutupandshave** [peterhaneve/evemon#54]
Sorry, me again :)

I've added about 60 chars in prep for your update and now it's crashing on load with red crosses suddenly appearing over some of the accounts (I can try and get you a SS if it helps).  I don't want to re-add them all again.  Is there 1) a limit and 2) any way of fixing the database?

**Comment by shutupandshave** [comment]
It seems to die when the settings file gets over 1.6gb

**Comment by shutupandshave** [comment]
I've managed to fix my settings.xml but I cant add any more chars atm.

**Comment by peterhaneve** [comment]
While it does slow down EVEMon, I was unable to reproduce your issue with 20 characters purposely inflated with large skill plans to make the settings file go well over 2 MB. Do you have a sample settings xml that fails to load?

**Comment by shutupandshave** [comment]
Can you give me an email address to send this too.  It's relatively sensitive info :)

**Comment by peterhaneve** [comment]
Is your settings file obviously truncated in the middle?

**Comment by shutupandshave** [comment]
No, I had a look through.  Can you ping me here (or in game) an email address I can send this file too?
KIATolon or Shutupandshave are good people to let me know who to send the file too?

**Comment by peterhaneve** [comment]
You can use peterhaneve@outlook.com as it is already listed in each of my commits.

**Comment by peterhaneve** [comment]
After looking at your settings file, it appears to be crashing when creating new tabs. The debug message is that EVEMon cannot create any more GUI window handles (controls), and it looks like EVEMon may have exceeded the [maximum number of window handles allowed in a process](https://stackoverflow.com/questions/522534/what-to-do-about-error-creating-window-handle-errors-in-a-c-sharp-application) with that many characters.

While possible to fix, changing EVEMon to use fewer controls is likely to require a big rewrite. One potential workaround would be to run multiple instances of EVEMon with the APPDATA folder redirected, using a procedure similar to [this page](https://ccm.net/faq/9814-portable-minecraft-minecraft-on-your-usb-key).

**Comment by shutupandshave** [comment]
Thanks a lot for investigating this.  I can probably get some more windows by not having pointless alts there, however having multiple profiles, whilst not ideal is certainly a reasonable workaround.  

 

Thanks again.
